<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>c_gradation</class>
 <widget class="QDialog" name="c_gradation">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>491</width>
    <height>389</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QWidget" name="layoutWidget_2">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>10</y>
     <width>77</width>
     <height>371</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QPushButton" name="b_ok">
      <property name="text">
       <string>确认</string>
      </property>
      <property name="autoDefault">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="b_cancel">
      <property name="text">
       <string>取消</string>
      </property>
      <property name="autoDefault">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QGroupBox" name="gbox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>381</width>
     <height>361</height>
    </rect>
   </property>
   <property name="title">
    <string>调整色阶</string>
   </property>
   <widget class="QWidget" name="widget_2" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>351</width>
      <height>221</height>
     </rect>
    </property>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>60</width>
       <height>16</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>输入色阶：</string>
     </property>
    </widget>
    <widget class="QLabel" name="l_hist">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>331</width>
       <height>150</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>hist</string>
     </property>
    </widget>
    <widget class="QWidget" name="widget_4" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>190</y>
       <width>351</width>
       <height>41</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="2">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="3">
       <widget class="QSpinBox" name="input_gray">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>65534</number>
        </property>
        <property name="singleStep">
         <number>100</number>
        </property>
        <property name="value">
         <number>32767</number>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="input_black">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximum">
         <number>65533</number>
        </property>
        <property name="singleStep">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="5">
       <widget class="QSpinBox" name="input_white">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <number>2</number>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="singleStep">
         <number>100</number>
        </property>
        <property name="value">
         <number>65535</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="myslider" name="inb">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>180</y>
       <width>13</width>
       <height>13</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, stop:0 rgba(0, 0, 0, 255), stop:1 rgba(100, 100, 100, 255));
border: 0px solid #3A3939;
color: silver;

    border-width: 1px;
    border-color: #4A4949;
    border-style: solid;
    padding-top: 5px;
    padding-bottom: 5px;
    padding-left: 5px;
    padding-right: 5px;
    border-radius: 2px;
    outline: none;</string>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="gstart" stdset="0">
      <number>40</number>
     </property>
     <property name="gend" stdset="0">
      <number>360</number>
     </property>
     <property name="lstart" stdset="0">
      <number>10</number>
     </property>
     <property name="lend" stdset="0">
      <number>169</number>
     </property>
     <property name="lmax" stdset="0">
      <number>328</number>
     </property>
     <property name="lmin" stdset="0">
      <number>10</number>
     </property>
     <property name="pos" stdset="0">
      <number>10</number>
     </property>
     <property name="start" stdset="0">
      <number>40</number>
     </property>
     <property name="tend" stdset="0">
      <number>330</number>
     </property>
    </widget>
    <widget class="myslider" name="inw">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>180</y>
       <width>13</width>
       <height>13</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: qlineargradient(spread:pad, x1:1, y1:0.460409, x2:0, y2:0.420545, stop:0 rgba(230, 230, 230, 255), stop:1 rgba(170, 170, 170, 255));
color: silver;
border: 0px solid #3A3939;
    border-color: #4A4949;
    border-style: solid;
    padding-top: 5px;
    padding-bottom: 5px;
    padding-left: 5px;
    padding-right: 5px;
    border-radius: 2px;
    outline: none;</string>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="gstart" stdset="0">
      <number>40</number>
     </property>
     <property name="gend" stdset="0">
      <number>360</number>
     </property>
     <property name="lstart" stdset="0">
      <number>171</number>
     </property>
     <property name="lend" stdset="0">
      <number>330</number>
     </property>
     <property name="lmax" stdset="0">
      <number>330</number>
     </property>
     <property name="lmin" stdset="0">
      <number>12</number>
     </property>
     <property name="pos" stdset="0">
      <number>330</number>
     </property>
     <property name="start" stdset="0">
      <number>40</number>
     </property>
     <property name="tend" stdset="0">
      <number>330</number>
     </property>
    </widget>
    <widget class="myslider" name="ing">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>180</y>
       <width>13</width>
       <height>13</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: qlineargradient(spread:pad, x1:1, y1:0.460409, x2:0, y2:0.420545, stop:0 rgba(170, 170, 170, 255), stop:1 rgba(90, 90, 90, 255));
color: silver;
border: 0px solid #3A3939;
    border-color: #4A4949;
    border-style: solid;
    padding-top: 5px;
    padding-bottom: 5px;
    padding-left: 5px;
    padding-right: 5px;
    border-radius: 2px;
    outline: none;</string>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="gstart" stdset="0">
      <number>40</number>
     </property>
     <property name="gend" stdset="0">
      <number>360</number>
     </property>
     <property name="lstart" stdset="0">
      <number>11</number>
     </property>
     <property name="lend" stdset="0">
      <number>329</number>
     </property>
     <property name="lmax" stdset="0">
      <number>328</number>
     </property>
     <property name="lmin" stdset="0">
      <number>10</number>
     </property>
     <property name="pos" stdset="0">
      <number>170</number>
     </property>
     <property name="start" stdset="0">
      <number>40</number>
     </property>
     <property name="tend" stdset="0">
      <number>330</number>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>260</y>
      <width>351</width>
      <height>101</height>
     </rect>
    </property>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>61</width>
       <height>19</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>输出色阶：</string>
     </property>
    </widget>
    <widget class="QLabel" name="l_bar">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>331</width>
       <height>21</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>bar</string>
     </property>
    </widget>
    <widget class="QWidget" name="widget_5" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>60</y>
       <width>351</width>
       <height>41</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QSpinBox" name="output_black">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>65534</number>
        </property>
        <property name="singleStep">
         <number>100</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QSpinBox" name="outpu_white">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="singleStep">
         <number>100</number>
        </property>
        <property name="value">
         <number>65535</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="myslider" name="outb">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>13</width>
       <height>13</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, stop:0 rgba(0,0, 0, 255), stop:1 rgba(130, 130, 130, 255));
color: silver;

border: 0px solid #3A3939;
    border-color: #4A4949;
    border-style: solid;
    padding-top: 5px;
    padding-bottom: 5px;
    padding-left: 5px;
    padding-right: 5px;
    border-radius: 2px;
    outline: none;</string>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="gstart" stdset="0">
      <number>40</number>
     </property>
     <property name="gend" stdset="0">
      <number>360</number>
     </property>
     <property name="lstart" stdset="0">
      <number>10</number>
     </property>
     <property name="lend" stdset="0">
      <number>329</number>
     </property>
     <property name="lmax" stdset="0">
      <number>328</number>
     </property>
     <property name="lmin" stdset="0">
      <number>10</number>
     </property>
     <property name="pos" stdset="0">
      <number>10</number>
     </property>
     <property name="start" stdset="0">
      <number>40</number>
     </property>
     <property name="tend" stdset="0">
      <number>330</number>
     </property>
    </widget>
    <widget class="myslider" name="outw">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>50</y>
       <width>13</width>
       <height>13</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: qlineargradient(spread:pad, x1:1, y1:0.460409, x2:0, y2:0.420545, stop:0 rgba(230, 230,230, 255), stop:1 rgba(130, 130, 130, 255));
color: silver;
border: 0px solid #3A3939;
    border-color: #4A4949;
    border-style: solid;
    padding-top: 5px;
    padding-bottom: 5px;
    padding-left: 5px;
    padding-right: 5px;
    border-radius: 2px;
    outline: none;</string>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="gstart" stdset="0">
      <number>40</number>
     </property>
     <property name="gend" stdset="0">
      <number>360</number>
     </property>
     <property name="lstart" stdset="0">
      <number>11</number>
     </property>
     <property name="lend" stdset="0">
      <number>330</number>
     </property>
     <property name="lmax" stdset="0">
      <number>328</number>
     </property>
     <property name="lmin" stdset="0">
      <number>10</number>
     </property>
     <property name="pos" stdset="0">
      <number>330</number>
     </property>
     <property name="start" stdset="0">
      <number>40</number>
     </property>
     <property name="tend" stdset="0">
      <number>330</number>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>myslider</class>
   <extends>QPushButton</extends>
   <header>myslider.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
